tests/functional/test_report_only_no_errors.py

```diff
@@ -0,0 +1,319 @@
+def test_partner_submits_report_without_any_errors():
+    """The partner hashicorp submits a error-free report for the vault chart."""
+
+
+@scenario('features/report_without_chart.feature', "The redhat associate submits a error-free report for the vault chart")
+def test_redhat_submits_report_without_any_errors():
+    """The redhat associate submits a error-free report for the vault chart."""
+
+
+@given("hashicorp is a valid partner")
+def hashicorp_is_a_valid_partner(secrets):
+    secrets.vendor_type = 'partners'
+    secrets.vendor = 'hashicorp'
+
+
+@given("a redhat associate has a valid identity")
+def hashicorp_is_a_valid_partner(secrets):
+    secrets.vendor_type = 'redhat'
+    secrets.vendor = 'redhat'
+
+
+@given("hashicorp has created a error-free report")
+@given("the redhat associate has created a report without any errors")
+def the_user_has_created_a_report_without_errors(secrets):
+    """The user has created a report without any errors."""
+
+    repo = git.Repo(os.getcwd())
+    if os.environ.get('WORKFLOW_DEVELOPMENT'):
+        logger.info("Wokflow development enabled")
+        repo.git.add(A=True)
+        repo.git.commit('-m', 'Checkpoint')
+
+    headers = {'Accept': 'application/vnd.github.v3+json',
+               'Authorization': f'Bearer {secrets.bot_token}'}
+
+    # Get SHA from 'dev-gh-pages' branch
+    logger.info(
+        f"Create '{secrets.test_repo}:{secrets.pr_base_branch}-gh-pages' from '{secrets.test_repo}:dev-gh-pages'")
+    r = requests.get(
+        f'https://api.github.com/repos/{secrets.test_repo}/git/ref/heads/dev-gh-pages', headers=headers)
+    j = json.loads(r.text)
+    sha = j['object']['sha']
+
+    # Create a new gh-pages branch for testing
+    data = {'ref': f'refs/heads/{secrets.pr_base_branch}-gh-pages', 'sha': sha}
+    r = requests.post(
+        f'https://api.github.com/repos/{secrets.test_repo}/git/refs', headers=headers, json=data)
+
+    # Make PR's from a temporary directory
+    old_cwd = os.getcwd()
+    dr = tempfile.mkdtemp(prefix='tci-')
+    logger.info(f'Worktree directory: {dr}')
+    repo.git.worktree('add', '--detach', dr, f'HEAD')
+
+    os.chdir(dr)
+    repo = git.Repo(dr)
+    pathlib.Path(
+        f'charts/{secrets.vendor_type}/{secrets.vendor}/{secrets.chart_name}/{secrets.chart_version}').mkdir(parents=True, exist_ok=True)
+
+    # Create the OWNERS file from the string template
+    values = {'bot_name': secrets.bot_name,
+              'vendor': secrets.vendor, 'chart_name': secrets.chart_name}
+    content = Template(secrets.owners_file_content).substitute(values)
+    with open(f'charts/{secrets.vendor_type}/{secrets.vendor}/{secrets.chart_name}/OWNERS', 'w') as fd:
+        fd.write(content)
+
+    # Push OWNERS file to the test_repo
+    logger.info(
+        f"Push OWNERS file to '{secrets.test_repo}:{secrets.pr_base_branch}'")
+    repo.git.add('charts')
+    repo.git.commit(
+        '-m', f"Add {secrets.vendor} {secrets.chart_name} OWNERS file")
```

Comment:
I'm warming up to using f-strings as well: https://realpython.com/python-f-strings/ 
@mmulholla ^^^^